src/reader/reader.rs

```diff
@@ -6646,21 +6653,32 @@ impl<'a> Reader<'a> {
 
             // Only use completions that match replace_token.
             let completion_replaces_token = c.flags.contains(CompleteFlags::REPLACES_TOKEN);
+            let needs_case_fix = completion_needs_case_fix(c);
             if completion_replaces_token != will_replace_token {
-                continue;
+                // Keep smart/samecase results even if we prefer not to replace the token.
+                if !needs_case_fix || will_replace_token {
+                    continue;
+                }
             }
 
             // Don't use completions that want to replace, if we cannot replace them.
             if completion_replaces_token && !reader_can_replace(&tok, c.flags) {
                 continue;
             }
 
-            // This completion survived.
-            surviving_completions.push(c.clone());
-            all_matches_exact_or_prefix =
-                all_matches_exact_or_prefix && c.r#match.is_exact_or_prefix();
+            all_matches_exact_or_prefix &= c.r#match.is_exact_or_prefix();
+
+            let mut completion = c.clone();
+            if needs_case_fix && !will_replace_token {
+                completion.flags |= CompleteFlags::SUPPRESS_PAGER_PREFIX;
```

Comment:
there was a rendering glitch - if I complete "cd do",
then the completion candidate for "Dodo" in the completion pager
will have two trailing spaces highlihted with `fish_pager_color_selected_background`.

Fixed by updating computation of `comp_width` to honor `SUPPRESS_PAGER_PREFIX`.